WooCommerce/Classes/ViewRelated/Blaze/BlazeEditAd/BlazeEditAdViewModel.swift

```diff
@@ -0,0 +1,204 @@
+import Foundation
+import Combine
+
+/// View model for `BlazeEditAdView`
+final class BlazeEditAdViewModel: ObservableObject {
+    typealias ImageState = EditableImageViewState
+
+    let siteID: Int64
+    private let adData: BlazeEditAdData
+
+    // Image selection
+    var onAddImage: ((MediaPickingSource) async -> MediaPickerImage?)?
+    @Published private(set) var imageState: ImageState
+    var image: MediaPickerImage? {
+        imageState.image
+    }
+
+    // Tagline
+    @Published var tagline: String = ""
+    var taglineFooterText: String {
+        taglineEmptyError ?? taglineLengthLimitLabel
+    }
+    private let taglineMaxLength = 32
+    @Published private var taglineRemainingLength: Int
+    private var taglineLengthLimitLabel: String {
+        String(format: Localization.taglineLengthLimit, taglineRemainingLength)
```

Comment:
We should use `String.pluralize` to update the text correctly when the remaining length is 1 or 0.

Reply:
Good suggestion. Done in 72a88cec7152cf4a946012e66615e9e2b73df0ee

I also started using the same localized string for both tagline and description fields.